File: src/things/LightOrb.cpp
#include "things/LightOrb.h"

LightOrb::LightOrb(UUID id) : Thing("light_orb", id)
{
}

LightOrb::~LightOrb() {}

std::string LightOrb::_get_description() const
{
  return "test lighting orb";
}

sf::Vector2u LightOrb::get_tile_sheet_coords(int frame) const
{
  return sf::Vector2u(1, 8);
}

File: resources/things/LightOrb.lua
-- Definition of the LightOrb object type.

require "resources/things/LightSource"

LightOrb = inheritsFrom(LightSource, "LightOrb")
LightOrb.intrinsics.name = "magical lighting orb"
LightOrb.intrinsics.plural = "magical lighting orbs"
LightOrb.intrinsics.creatable = true

LightOrb.intrinsics.light_color_red = 255
LightOrb.intrinsics.light_color_green = 255
LightOrb.intrinsics.light_color_blue = 255
LightOrb.intrinsics.light_level = 256
LightOrb.intrinsics.light_strength = 255

LightOrb.intrinsics.physical_mass = 2

function LightOrb.get_description()
	return "A glowing orb, used for testing."
end